nimbus/src/updating.rs

```diff
@@ -0,0 +1,72 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! This module implements the primitive functions to implement
+//! safe updating from the server.
+
+use crate::error::Result;
+use crate::persistence::{Database, StoreId, Writer};
+use crate::Experiment;
+
+const KEY_PENDING_UPDATES: &str = "pending-experiment-updates";
+
+pub fn write_pending_experiments(db: &Database, experiments: Vec<Experiment>) -> Result<()> {
+    let mut writer = db.write()?;
+    db.get_store(StoreId::Updates)
+        .put(&mut writer, KEY_PENDING_UPDATES, &experiments)?;
+    writer.commit()?;
+    Ok(())
+}
+
+pub fn read_and_remove_pending_experiments(
+    db: &Database,
+    writer: &mut Writer,
+) -> Result<Option<Vec<Experiment>>> {
+    let store = db.get_store(StoreId::Updates);
+    let experiments = store.get::<Vec<Experiment>>(writer, KEY_PENDING_UPDATES)?;
+
+    if experiments.is_some() {
+        store.clear(writer)?;
+    }
+
+    Ok(experiments)
+}
+
+#[cfg(feature = "rkv-safe-mode")]
```

Comment:
I guess this test also crashes rkv for no good reason? If so, please add a comment indicating this (and if not, then please remove it :)